src/vos/vos_io.c

```diff
@@ -203,8 +209,10 @@ vos_ioc_create(daos_handle_t coh, daos_unit_oid_t oid, bool read_only,
 		int iov_nr = iods[i].iod_nr;
 		struct bio_sglist *bsgl;
 
-		if (iods[i].iod_type == DAOS_IOD_SINGLE && iov_nr != 1) {
-			D_ERROR("Invalid sv iod_nr=%d\n", iov_nr);
+		if ((iods[i].iod_type == DAOS_IOD_SINGLE && iov_nr != 1) ||
+		    iov_nr <= 0) {
```

Comment:
@daos_iod_t::iod_nr is "unsigned int", it is assigned to "int" local variable @iov_nr, so in theory, the "iov_nr <= 0" may cause unexpected wrong result if the real @daos_iod_t::iod_nr is greater than (1 << 31). So we may need to check "if (iov_nr == 0)" ?

Reply:
ok, thx